contrib/opencensus-ext-azure/opencensus/ext/azure/common/__init__.py

```diff
@@ -17,14 +17,76 @@
 
 from opencensus.ext.azure.common.protocol import BaseObject
 
+AUTHORIZATION = 'Authorization'
+DEFAULT_BREEZE_ENDPOINT = 'https://dc.services.visualstudio.com'
+ENDPOINT_SUFFIX = 'EndpointSuffix'
+ENV_CONNECTION_STRING = 'APPLICATIONINSIGHTS_CONNECTION_STRING'
+ENV_INSTRUMENTATION_KEY = 'APPINSIGHTS_INSTRUMENTATIONKEY'
+INGESTION_ENDPOINT = 'IngestionEndpoint'
+INSTRUMENTATION_KEY = 'InstrumentationKey'
+LOCATION = 'Location'
+
+
+def process_options(options):
+    code_cs = parse_connection_string(options.connection_string)
+    code_ikey = options.instrumentation_key
+    env_cs = parse_connection_string(os.getenv(ENV_CONNECTION_STRING))
+    env_ikey = os.getenv(ENV_INSTRUMENTATION_KEY)
+
+    options.instrumentation_key = code_cs.get(INSTRUMENTATION_KEY) \
+        or code_ikey \
+        or env_cs.get(INSTRUMENTATION_KEY) \
+        or env_ikey
+    endpoint = code_cs.get(INGESTION_ENDPOINT) \
+        or env_cs.get(INGESTION_ENDPOINT) \
+        or DEFAULT_BREEZE_ENDPOINT
+    options.endpoint = endpoint + '/v2/track'
+
+
+def parse_connection_string(connection_string):
+    if connection_string is None:
+        return {}
+    try:
+        pairs = connection_string.split(';')
+        result = dict(s.split('=') for s in pairs)
+    except Exception:
+        raise ValueError('Invalid connection string: ' + connection_string)
+    # Validate authorization
+    auth = result.get(AUTHORIZATION)
+    if auth is None:
+        raise ValueError('Missing \'Authorization\' in connection string:'
```

Comment:
This can be a valid connection string as per
https://github.com/microsoft/ApplicationInsights-dotnet/pull/1187/files#diff-7421a28dc03eceb294044ed77594ec83R12-R18

Reply:
The initial version of the Connection String spec mentions that Authorization is optional because today we only support "ikey" based auth.

Reply:
Sounds good.